app/waf/src/controllers/application.controller.ts

```diff
@@ -155,4 +173,79 @@ export class ApplicationController {
   async deleteById(@param.path.string('id') id: string): Promise<void> {
     await this.applicationRepository.deleteById(id);
   }
+
+  @post(prefix + '/applications/{id}/deploy', {
+    responses: {
+      '204': {
+        description: 'Application deploy success',
+      },
+    },
+  })
+  async deployById(@param.path.string('id') id: string): Promise<Object> {
+    let application = await this.applicationRepository.findById(id);
+
+    let params: {[key: string]: Object} = {
+      application: application,
+    };
+
+    let tenantAssoc = null;
+    try {
+      tenantAssoc = await this.tenantAssociationRepository.findById(
+        application.tenantId,
+      );
+    } catch (e) {
+      if (e.code !== 'ENTITY_NOT_FOUND') {
+        throw e;
+      }
+    }
+
+    if (!tenantAssoc) {
+      try {
+        tenantAssoc = await this.tenantAssociationRepository.findById(
+          'default',
+        );
+      } catch (e) {
+        if (e.code === 'ENTITY_NOT_FOUND') {
+          throw new HttpErrors.UnprocessableEntity(
+            'No ADC associated with Application ' + application.id,
+          );
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    let adc = await this.adcRepository.findById(tenantAssoc.adcId);
```

Comment:
Use `find` or you need to `try...catch` the possible exception.

Reply:
Done